src/context.cpp

```diff
@@ -711,158 +726,103 @@ namespace librealsense
             software_sensor::open( profiles );
         }
 
-        void custom_on_video_frame( rs2_software_video_frame const & software_frame )
+        void handle_video_data( realdds::topics::device::image && dds_frame,
+                                const std::shared_ptr< stream_profile_interface > & profile,
+                                frame_metadata_syncer & syncer )
         {
-            // We do exactly the same as the base on_video_frame(), but we have a custom releaser that know that the
-            // pixels are actually pointing to a vector:
-            auto p_pixels_vector = static_cast< std::vector< byte > * >( software_frame.pixels );
-            rsutils::deferred on_release( [p_pixels_vector]() { delete p_pixels_vector; } );
+            frame_additional_data data;  // with NO metadata by default!
+            data.timestamp; // from metadata
+            data.timestamp_domain; // from metadata
+            data.depth_units; // from metadata
+            data.frame_number = ! dds_frame.frame_id.empty() ? std::stoi( dds_frame.frame_id ) : 0;
```

Comment:
`stoi` might throw

Reply:
Changed both (video/motion) to set to 0 if invalid